### PR TITLE
Created a reusable "Drawer" component

### PR DESCRIPTION
**Pull request by uskayyyyy**
## Overviews
Created a reusable "Drawer" component and its usage example file.

## Review Points
- Focused on designing the drawer of the image below
image

- Remove styling the green background drawers from the scope of this PR
Screenshot 2023-12-13 at 16 21 52

## Screenshots

- The body is scrollable when the content exceeds the parent height
- The overlay remains of "2rem"

https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/9a3ca72b-3020-4eec-aeef-f34fdbe7d13f

- If the body height is lower than its parent, the height of the entire drawer shrinks to fit the content

https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/22cdc27d-3c3b-42d4-8cca-eecbb7ecbe0a

- The Drawer can be set to the right side
- The size of the right drawer may need to be adjusted when applying it

https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/2467741e-9c51-459a-80f8-01f19d5c8333

### CODE REVIEW

File: src/components/ui/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+const DrawerTrigger = PrimitiveTrigger;
+const DrawerClose = PrimitiveClose;
+const DrawerPortal = PrimitivePortal;
+
+const DrawerOverlay = forwardRef<
+  ElementRef<typeof PrimitiveOverlay>,
+  ComponentPropsWithoutRef<typeof PrimitiveOverlay>
+>(({ className, ...props }, ref) => (
+  <PrimitiveOverlay
+    className={cn(
```
Comment (kanta1207) [inline src/components/ui/Drawer.tsx:37]: I liked how you classify className by it's categories!
Maybe we can create a note or doc somewhere about how we classify and organize classNames in cn() method

**Comment by kanta1207** [comment]
LGTM! 
Tried to find a room for improvement but couldn't! 
Beautifully implemented 👏

**Review by kanta1207 — APPROVED** [review]

File: src/components/ui/Drawer.tsx (hunk 1 above)
Comment (uskayyyyy) [reply src/components/ui/Drawer.tsx:37]: I was considering creating a `STYLEGUIDE.md` file to document coding rules for a project, such as using `I` to name interfaces like `IMyProps`.

Comment (uskayyyyy) [reply src/components/ui/Drawer.tsx:37]: Created #91

**Comment by uskayyyyy** [comment]
@kanta1207 
Thank you for your review! I will wait for Kota's review, just in case.

File: src/components/ui/Dialog.tsx [hunk 2]
```diff
@@ -79,6 +79,16 @@ const DialogHeader = ({ className, ...props }: HTMLAttributes<HTMLDivElement>) =
 );
 DialogHeader.displayName = 'DialogHeader';
 
+/**
+ * This is a custom component that we use to wrap the body (between header and footer) of the dialog.
+ */
+const DialogBody = forwardRef<ElementRef<'div'>, HTMLAttributes<HTMLDivElement>>(
```
Comment (kotaaaa) [inline src/components/ui/Dialog.tsx:85]: Do you still need this `DialogBody`?

Comment (kotaaaa) [reply src/components/ui/Dialog.tsx:85]: Oh, Yes, don't care about this comment

**Review by kotaaaa — APPROVED** [review]
Thank you for implementation!!!

File: src/components/ui/Dialog.tsx (hunk 2 above)
Comment (uskayyyyy) [reply src/components/ui/Dialog.tsx:85]: 😆

**Pull request merged**